text/3730-nit-lint-level.md

```diff
@@ -0,0 +1,379 @@
+- Feature Name: `nit-lint-level`
+- Start Date: 2024-11-15
+- RFC PR: [rust-lang/rfcs#3730](https://github.com/rust-lang/rfcs/pull/3730)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- [Internals](https://internals.rust-lang.org/t/forbid-deny-warn-allow-and-notice/19986)
+
+# Summary
+[summary]: #summary
+
+Add a new visible lint level below `warn` to allow linters to act like a pair-programmer / code-reviewer where feedback is evaluated on a case-by-case basis.
+- `cargo` does not display these lints by default, requiring an opt-in
+- This RFC assumes LSPs/IDEs will opt-in
+- CIs could opt-in and open issues in the code review to show nits introduced in the current PR but not block merge on them
+
+There is no expectation that crates will be `nit`-free.
+
+*Note:* The name `nit` is being used for illustrative purposes and is assumed to not be the final choice
+
+*Note:* This RFC leaves the determination of what lints will be `nit` by
+default to the respective teams.  Any lints referenced in this document are
+for illustrating the intent of this feature and how teams could reason about
+this new lint level.
+
+# Motivation
+[motivation]: #motivation
+
+By its name and literal usage, the `warn` level is non-blocking.
+However, most projects treat `warn` as a soft-error.
+It doesn't block for local development but CI blocks it from being merged.
+This is an attempt to balance final correctness with rapid prototyping.
+Requiring "warnings clean" code also avoids warnings fatigue where warnings
+make it hard to see "relevant' compiler output and
+act as proverbial [broken windows](https://en.wikipedia.org/wiki/Broken_windows_theory).
```

Comment:
```suggestion
Requiring "warnings clean" code also avoids warnings fatigue where warnings
make it hard to see "relevant" compiler output, and make a codebase feel
lower-quality in a way that does not inspire people or invite people to
help solve the problem.
```

(Let's avoid the substantial political connotations of the phrase, and convey the underlying concept in a self-explanatory fashion instead.)